z390test/README.md

```diff
@@ -0,0 +1,239 @@
+assert rc == 0
+```
+Parameters:
+* asmFile - The HLASM file without prefix to assemble. Use basePath method to build path relative to z390.
+* args... - 0-n parmeters to pass to the asm program, generally options.
+
+Returns:
+* rc - The result code from the execution
+
+Methods available :
+* asm - Assemble only
+* asml - Assemble and link. Any non zero RC on assemble will exit with RC
+* asmlg - Assemble, link and go. Any non zero RC on assemble and link will exit with RC
+* mz390 - Assemble only
+* cblc - COBOL assemble only
+* cblclg - COBOL assemble, link and go. Any non-zero return will exist with RC
+* lz390 - Link only. Output not cleared prior to run
+* ez390 - Run only. Output not cleared prior to run
+
+### assemble from inline source
+
+```groovy
+var source = """TESTBR14 CSECT
+         SR    15,15
+         BR    14
+         END
+"""
+String sourceFilename = createTempFile("INLINE.MLC", source)
+int rc = this.asm(sourceFilename, *options)
+assert rc == 0
+```
+Use the `createTempFile` method to create a file that can be
+used in the assembly.
+
+### Set environment variables
+
+```groovy
+this.env = ['SNAPOUT': basePath('zopcheck', 'SNAPOUT.TXT')]
+```
+The env property is a set of key value pairs within a Java or [Groovy map](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html). 
+These will be passed as environment variables for the z390 calls.
+
+To add a single envvar without replacing any previously set environment variables:
+```groovy
+this.env.put('SNAPOUT', basePath('zopcheck', 'SNAPOUT.TXT'))
+```
+
+
+### loadFile -- loads content of a file into fileData
+
+```groovy
+loadFile(filename, label)
+```
+* filename - is the file that will be loaded into the fileData area.
+* label - the key in the fileData hashlist
+
+You can access the file contents after loading the file
+
+```groovy
+fileData[label] 
+```
+
+### printOutput -- prints captured data
+
+```groovy
+printOutput()
+```
+Prints the captured output to stdout - useful for debugging output.
+
+Captured output includes stdout, stderr, the assembly output files (LST, LOG, PRN and ERR files) 
+and any other files you load using the `loadFile` method.
+
+First 5 digits are line numbers starting at 0 are included in the listing.
+
+### basePath -- construct path relative to z390 source root
+
+```groovy
+basePath("tests", "TESTINS1")
+```
+
+Use this to construct file paths relative to the z390 source root. Ensures platform independent 
+file handling.
+
+Set the z390 source root using envvar `Z390_SOURCE_ROOT`. If not provided, defaults to "../z390"
+
+### createTempFile -- create a temp file for use in the test
+
+```groovy
+var tempFileContents = """Line 1: Some text
+Line 2: This is a test file
+"""
+var tempFilename = createTempFile("myTempFile.txt", tempFileContents)
+println(tempFilename)
+```
+
+Use this to create temp files used in your tests. Returns full path to temp filename.
+
+Note - Temp directory is deleted at end of testcase.
```

Comment:
Temp directory??? Do you mean the temp file?

Reply:
I do mean directory, but thats because the temp file is generated in a temp directory -- at least on Windows and MacOS.. I think Linux may just use the /tmp folder.. I will update to say that the temp file and any generated directory containing the temp file will be deleted.